pkg/envoy/sds/response.go

```diff
@@ -145,58 +144,66 @@ func (s *sdsImpl) getRootCert(cert certificate.Certificater, sdscert envoy.SDSCe
 		return secret, nil
 	}
 
-	// Program SAN matching based on SMI TrafficTarget policies
-	switch sdscert.CertType {
-	case envoy.RootCertTypeForMTLSOutbound:
+	svcIdentities, err := getServiceIdentitiesFromCert(sdsCert, s.serviceIdentity, s.meshCatalog)
```

Comment:
This function looks confusing because it passes in the proxy's own ServiceIdentity and returns a list of ServiceIdentity. Renaming the variable might clear the confusion.

```suggestion
	svcIdentitiesInCertRequest, err := getServiceIdentitiesFromCert(sdsCert, s.serviceIdentity, s.meshCatalog)
```